### ISSUE TITLE
Custom Extension on NPM - Schema Not Found error (v2.2.8)

### ISSUE THREAD

**Issue by TK-21st** [jupyterlab/jupyterlab#9216]
## Description
We're trying to publish [`FlyBrainLab/NeuroMynerva`](https://github.com/FlyBrainLab/NeuroMynerva) to NPM. The installation works fine (including inside JupyterLab's extension manager) but the schema cannot be found.

In development the schema works fine, it's just when published to NPM it's having problems.

I believe I've included the right files in our [package.json](https://github.com/FlyBrainLab/NeuroMynerva/blob/master/package.json).

## Reproduce

```
jupyter labextension install @flybrainlab/neuromynerva
```

After clicking on one of the buttons in the Launch under `FlyBrainLab` section,
I see that the schema is not found:

![image](https://user-images.githubusercontent.com/26802334/97041398-8300fb00-153d-11eb-8e29-a806ed76448d.png)

The schema folder is also not showing up in webpack:
![image](https://user-images.githubusercontent.com/26802334/97041444-93b17100-153d-11eb-9dd7-69f6cb14856c.png)

## Expected behavior
Schema should show up 😢 

## Context

- Operating System and version: Mac OS X 10.15.6
- Browser and version: Chrome 86.0.4240.111
- JupyterLab version: 2.2.8

**Comment by telamonian** [comment]
It's a pathing problem. The code here in your [index.ts](https://github.com/FlyBrainLab/NeuroMynerva/blob/2b31af8e0ee3fbb55e14a9e7690d3d14516f90dd/src/extension.ts#L171-L177):

```typescript
const extension: JupyterFrontEndPlugin<IFBLWidgetTrackers> = {
  id: '@neuro-mynerva/fbl-extension:plugin',
  autoStart: true,
  requires: [ICommandPalette, ILauncher, ILayoutRestorer, ISettingRegistry],
  provides: IFBLWidgetTrackers,
  activate: activateFBL
};
```

should have `id: '@flybrainlab/neuromynerva'`.

Generally my advice in these situations is to use [jupyterlab-git as a reference](https://github.com/jupyterlab/jupyterlab-git/blob/master/src/index.ts), since that's a well maintained extension project.

**Comment by TK-21st** [comment]
Yes! Thanks a ton @telamonian! I changed the package name in package.json without changing the code.. 😮

**Comment by telamonian** [comment]
For future reference, you need to specify an extension's settings schema in three places exactly:

- the schema file path in files in package.json
- the schema dir path in jupyterlab.schemaDir in package.json
- "<package-name>:<schema-file-name-without-extension>" as the id of your frontend extension

**Comment by ianhi** [comment]
I'm having a similar issue with a federated extension. As far as I can tell the name of the files is the same in all three place. I also included the schema file in the `MANIFEST.in` 
It works if I install with `pip install -e .` but
```
python setup.py bdist_wheel
pip install dist/<whatever version>.whl
```
results in this error: 
`Uncaught (in promise) Error: Schema not found: /home/ian/anaconda3/envs/burner/share/jupyter/lab/schemas/jupyterlab-vimrc/vimrc.json`

Is there anything extra that needs to be done in order to include the settings for a pip installable extension? The setup.py for 3.0 is pretty different so unfortunately I don't think I can figure this out via jlab-git.

For reference the branch is here: https://github.com/ianhi/jupyterlab-vimrc/tree/jlab3

**Comment by telamonian** [comment]
@ianhi I looked over your code. You have a different problem from the OP here, so you should open a new issue about it.

One way or another, your ultimate problem is that, while your schema files are getting included somewhere in your built .whl, they aren't getting installed into any of the jupyterlab application paths (not that I'm familiar enough with the new build system yet to know exactly where they're supposed to go)

**Comment by jasongrout** [comment]
Does this recent fix in the cookie cutter help?

https://github.com/jupyterlab/extension-cookiecutter-ts/pull/105

**Comment by telamonian** [comment]
@ianhi I arrived at Jason's suggestion independently and just tested it out locally, and that does seem to fix it. For the federated extensions, it appears that the settings schema .json needs to end up at something like:

```
<your-local-whatever>/share/jupyter/labextensions/<ext-name>/schemas/<ext-name>/<schema-file>
```

and changing your `data_files_spec` in setup.py to:

```python
data_files_spec = [
    ("share/jupyter/labextensions/%s" % labext_name, lab_path, "**"),
]
```

appears to be sufficient to make that happen

**Comment by ianhi** [comment]
ah indeed that fixes it. Thanks all!

I had ran the upgrade extension script yesterday so I missed out on that PR.
